### PR TITLE
Enabling ClamAV

### LINKED ISSUE #1913: File scanning using clamav module for uploaded files
## Title

Scan your files using clamav before uploading them (needs better title ...)
## Description

We've recently seen a number of tickets (#73955, #73939) in which suspicious files have been uploaded via fields. These ended up in the files/ directory and have caused alerts in some cases, harming site SEO and credibility.

Modules such as ClamAV (https://www.drupal.org/project/clamav) or Upload Scanner https://wordpress.org/plugins/upload-scanner/ leverage the clamav binary installed on Pantheon to scan every file that's being uploaded. Although this would slow down the upload process, it would provide better security and would ultimately be a good practice for every site.
## Expected Audience

Site administrators, developers.
## Path

source/docs/articles/sites/clamav
(e.g. `source/docs/articles/sites/code` or `source/docs/articles/wordpress`)

### PR DESCRIPTION
**Pull request by eabquina**
New docs on how to download and install ClamAV

Closes #1913 

## Effect
PR includes the following changes:
- New docs page about ClamAV

## Remaining Work
- [ ] List any outstanding todos
- [ ] If needed

### CODE REVIEW

**Comment by alexfornuto** [comment]
At this point I'd like to be able to confirm that the selected WordPress plugin will work with ClamAV without a daemon (ie in executable mode). I've successfully uploaded the eicar test signature with it installed, which worries me.

**Comment by alexfornuto** [comment]
image

It appears this may not work on Drupal either. More testing is needed, including testing by uploading clean files and known test signatures.

**Comment by eabquina** [comment]
I see that WordPress is working fine for ClamAV, but worried because the plugin is quite.
For ClamAV D7 and D8, nothing worked yet, will raise a BUG card now for ClamAV

**Comment by rvtraveller** [comment]
I can confirm that the ClamAV module is working on a D7 site on the platform using the EICAR test file.  I would also say though that recent infrastructure changes seem to have significantly slowed down ClamAV scanning.  The 68 byte test file takes 15+ seconds to scan on the new infrastructure and while I don't have numbers to prove it, I feel like I would have heard if the old infrastructure was taking that long.

**Comment by alexfornuto** [comment]
Blocked pending BUGS-1566

**Comment by alexfornuto** [comment]
The bug report is still pending.

**Comment by alexfornuto** [comment]
Bug report _still_ pending.

**Comment by alexfornuto** [comment]
Action on the bug report, it's being moved up in the backlog.

**Comment by alexdicianu** [comment]
@alexfornuto Updated this doc to reflect the new settings.

File: source/_docs/clamav.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    <li id="tab-1-id" role="presentation" class="active"><a href="#tab-1-anchor" aria-controls="tab-1-anchor" role="tab" data-toggle="tab">WordPress</a></li>
+    <!-- 2nd Tab Nav -->
+    <li id="tab-2-id" role="presentation"><a href="#tab-2-anchor" aria-controls="tab-2-anchor" role="tab" data-toggle="tab">Drupal</a></li>
+  </ul>
+
+<!-- Tab panes -->
+<div class="tab-content">
+<!-- Active pane content -->
+<div role="tabpanel" class="tab-pane active" id="tab-1-anchor" markdown="1">
+[Upload Scanner for WordPress](https://wordpress.org/plugins/upload-scanner){.external}
```
Comment (ari-gold) [inline source/_docs/clamav.md]: These instructions make it look like Terminus is required. Can we generalize the instructions?

File: source/_docs/clamav.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+<div role="tabpanel" class="tab-pane" id="tab-2-anchor" markdown="1">
+[ClamAV Module for Drupal 7 & 8](https://www.drupal.org/project/clamav){.external}
+
+1. Download the module via [Terminus](/docs/terminus/):
+
+   ```
+   terminus drush <site-name>.<env> -- dl clamav
+   ```
+
+2. Enable the module and set the scan method to daemon mode (from the module's configuration page, set the hostname and port number to localhost and 3310):
```
Comment (ari-gold) [inline source/_docs/clamav.md]: Maybe break the hostname and port number into separate step?

File: source/_docs/clamav.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+description: Learn how to protect your site against malicious files with malware and other potential threats using ClamAV, an open source antivirus engine.
+tags: [clamav, security, files]
+categories: [security]
+---
+
+## What is ClamAV?
+
+**ClamAV** is an open source antivirus engine for detecting trojans, viruses, malware & other malicious threats. It supports multiple file formats, file and archive unpacking, and multiple signature languages.
+
+A file that may look like a normal JPG or PDF file may actually be malware that run scripts within your site. They may come from files that have been uploaded via fields, attachments or other media forms. These may end up in your public files directory and might cause alerts, harm site SEO scores, compromise data, and or lower site credibility.
```
Comment (ari-gold) [inline source/_docs/clamav.md]: new section heading: "Why use ClamAV?" ?

File: source/_docs/clamav.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Pantheon servers have the ClamAV binary installed already. There are modules or plugins available for WordPress and Drupal that will use the binary. Once active, it will scan every file uploaded. Although this would slow down the upload process, the benefit of better security makes it a good practice for every site.
```
Comment (ari-gold) [inline source/_docs/clamav.md]: Suggest removing this paragraph

File: source/_docs/clamav.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+## Install ClamAV
+
+Because the ClamAV binary is already installed on Pantheon servers, all you need to do is install the available modules and plugins.
```
Comment (ari-gold) [inline source/_docs/clamav.md]: and plugins -> or plugins . ?

**Comment by ari-gold** [comment]
Product work complete, so pending any copy editing, this can be published.

File: source/_docs/clamav.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+---
+title: Enabling ClamAV for WordPress or Drupal
+description: Learn how to protect your site against malicious files with malware and other potential threats using ClamAV, an open source antivirus engine.
+tags: [clamav, security, files]
+categories: [security]
+---
+
+## What is ClamAV?
```
Comment (ari-gold) [inline source/_docs/clamav.md]: Suggest removing section heading

**Comment by alexfornuto** [comment]
@ari-gold thanks for the heads up - I still need to complete a technical review myself before this can go to copy editing.

File: source/_docs/clamav.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+**ClamAV** is an open source antivirus engine for detecting trojans, viruses, malware & other malicious threats. It supports multiple file formats, file and archive unpacking, and multiple signature languages.
```
Comment (ari-gold) [inline source/_docs/clamav.md]: suggest editing this paragraph down to a brief 1-sentence overview, something like:

**ClamAV** is an open source antivirus engine your WordPress or Drupal site can use to scan user-uploaded files.

File: source/_docs/clamav.md (hunk 3 above)
Comment (ari-gold) [inline source/_docs/clamav.md]: does WordPress/Drupal refuse to upload files that are malicious?

File: source/_docs/clamav.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+   ```
+   terminus drush <site-name>.<env> -- dl clamav
+   ```
+
+2. Enable the module and set the scan method to daemon mode (from the module's configuration page, set the hostname and port number to localhost and 3310):
+
+    ```
+    terminus drush <site-name>.<env> -- en clamav -y
+    terminus drush <site-name>.<env> -- vset clamav_mode 0
```
Comment (alexfornuto) [inline source/_docs/clamav.md]:
```
$ terminus drush $site.$env -- vset clamav_mode 0
Command variable-set requires Drupal core version 6 or 7 to run.         [error]
The drush command 'vset clamav_mode 0' could not be executed.            [error]
 [notice] Command: megasite-d8.dev -- drush vset clamav_mode 0 [Exit: 1]
 [error]   
```

File: source/_docs/clamav.md (hunk 3 above)
Comment (ari-gold) [reply source/_docs/clamav.md]: what happens when ClamAV detects a mailicious file? It just logs? or prevents upload?

Comment (alexfornuto) [reply source/_docs/clamav.md]: Oh, that depends on a config setting, which defaults to "block" - at least in the D8 module. Haven't got the WP plugin to work yet.

**Comment by alexfornuto** [comment]
Tested with eicar test file on D8, unable to verify through log files that ClamAV is running.
Unable to test on WordPress without further information on system commands required. (messaged @alexdicianu with details).

**Comment by alexfornuto** [comment]
@ari-gold @alexdicianu I _think_ it's finally ready. I'll wait until tomorrow in case you have any other comments.

**Pull request merged**